### PR TITLE
Add --noglob cli option

### LINKED ISSUE #4193: Allow to use full file path without converting it to glob
> What is the problem you're trying to solve?

I have been trying to add stylelint to a project. However, when our build bot runs our tests, it runs them in a directory path that happens to contain parentheses. This causes problems with globby, and leads to stylelint not actually linting anything. I tried running globby with `noext` set, but that just caused it to produce some arcane-looking errors.
I might have been configuring globby wrong, I suppose; in the end, though, I found it easiest to (write a PR that) disables globbing altogether. We weren't using globbing anyway.

> What solution would you like to see?

A command that disables globbing, and just makes stylelint assume whatever paths you provided are literal paths. Ideally a cli option, since we are invoking stylelint through the command line rather than through the javascript interface.

### PR DESCRIPTION
**Pull request by AzazKamaz**
Closes #4193 and is non-breaking alternative for #4254 that can be accepted in minor release.
Tests and flow types have been implemented

### CODE REVIEW

File: system-tests/cli/cli.test.js [hunk 1]
```diff
@@ -114,4 +114,46 @@ describe('CLI', () => {
 			);
 		});
 	});
+
+	it('--noglob-enabled-right', () => {
```
Comment (vankop) [inline system-tests/cli/cli.test.js]: Thanks for PR, it really useful feature. Could you please rename test cases without additional `-` and make more informative description. For example
```
describe('--no-glob', () => {
     it('enabled. file exists');
     it('enabled. file does not exist');
     it('disabled.  file exists');
     it('disabled. file does not exist');
})
```
```

**Comment by evilebottnawi** [comment]
I am :-1: on this, solution should be https://github.com/stylelint/stylelint/issues/4193#issuecomment-520214141, not new option for CLI

File: lib/standalone.js [hunk 2]
```diff
@@ -170,7 +170,7 @@ module.exports = function(
 		fileList = [fileList];
 	}
 
-	if (!options.disableDefaultIgnores) {
+	if (!options.disableDefaultIgnores && !options.noglob) {
```
Comment (evilebottnawi) [inline lib/standalone.js]: In this case, `ALWAYS_IGNORED_GLOBS` is not working as expected with `noglob`, it is not good idea

**Comment by vankop** [comment]
@evilebottnawi So, if I understand you correctly, you suggest to use kind of heuristic:
- if there is a file in file system using `options.files` as relative path, then we just lint this file
- otherwise expect globby pattern

or there is another idea?

**Comment by vankop** [comment]
eslint use [this code](https://github.com/eslint/eslint/blob/d08683e3c807f92daf266894093c70f8d5ac6afa/lib/cli-engine/file-enumerator.js#L292)

**Comment by evilebottnawi** [comment]
Code from eslint looks good, i think we should use this logic too

**Comment by vankop** [comment]
Any other suggestions?

**Pull request closed without merging**